resources/sdk/purecloudgo/templates/model.mustache

```diff
@@ -3,20 +3,77 @@ package {{packageName}}
 import ({{#imports}}
 	"{{import}}"{{/imports}}
 	"github.com/leekchan/timeutil"
+	"reflect"
 	"encoding/json"
 	"strconv"
 	"strings"
 )
 {{#model}}
 
 // {{classname}}{{#description}} - {{{description}}}{{/description}}
-type {{classname}} struct { {{#vars}}
+type {{classname}} struct { 
+	// SetFieldNames defines the list of fields to use for controlled JSON serialization
+	SetFieldNames map[string]bool `json:"-"`{{#vars}}
 	// {{name}}{{#description}} - {{{description}}}{{/description}}
 	{{name}} *{{{dataType}}} `json:"{{baseName}},omitempty"`
-
 {{/vars}}}
 
-func (o *{{classname}}) MarshalJSON() ([]byte, error) {
+// SetField uses reflection to set a field on the model if the model has a property SetFieldNames, and triggers custom JSON serialization logic to only serialize properties that have been set using this function.
+func (o *{{classname}}) SetField(field string, fieldValue interface{}) {
+	// Get Value object for field
+	target := reflect.ValueOf(o)
+	targetField := reflect.Indirect(target).FieldByName(field)
+
+	// Set value
+	if fieldValue != nil {
+		targetField.Set(reflect.ValueOf(fieldValue))
+	} else {
+		// Must create a new Value (creates **type) then get its element (*type), which will be nil pointer of the appropriate type
+		x := reflect.Indirect(reflect.New(targetField.Type()))
+		targetField.Set(x)
+	}
+
+	// Add field to set field names list
+	if o.SetFieldNames == nil {
+		o.SetFieldNames = make(map[string]bool)
+	}
+	o.SetFieldNames[field] = true
+}
+
+func (o {{classname}}) MarshalJSON() ([]byte, error) {
```

Comment:
I changed the struct attachment of the MarshalJSON field from `(o *{{classname}})` to just `(o {{classname}})`. The custom serializer wasn't getting called in my testing with the existing signature. I'm not sure why it was that way and if changing it will have any negative impacts. The SDK's tests still pass.